lib/iris/tests/unit/analysis/test_PERCENTILE.py

```diff
@@ -208,12 +208,25 @@ def test_masked(self):
         shape = (2, 11)
         data = ma.arange(np.prod(shape)).reshape(shape)
         data[0, ::2] = ma.masked
-        emsg = "Cannot use fast np.percentile method with masked array."
+        emsg = (
+            "Cannot use fast np.percentile method with masked array unless "
+            "mdtol is 0."
+        )
         with self.assertRaisesRegex(TypeError, emsg):
             PERCENTILE.aggregate(
                 data, axis=0, percent=50, fast_percentile_method=True
             )
 
+    def test_masked_mdtol_0(self):
+        shape = (2, 11)
```

Comment:
```suggestion
        shape = (3, 11)
```
I found that, with shape (2, 11), `np.percentile` returns a masked array anyway but doesn't with shape (3, 11).  Not sure why that should be.

Reply:
I'm not sure I understand? Is it that I'm not doing a sufficient test of my code masking the data because `np.percentile` will do it anyway for a (2, 11) so I need to use a (3, 11) to make sure my code is doing it?

Reply:
Yes, that's what I meant.  Sorry to be ambiguous.

Reply:
https://github.com/SciTools/iris/pull/4755/commits/34719085840786f74976468bec790bd6c5f76437